chromadb/api/models/CollectionCommon.py

```diff
@@ -510,35 +461,27 @@ def _process_upsert_or_update_request(
         )
 
         normalized_embeddings = (
-            self._normalize_embeddings(unpacked_embedding_set["embeddings"])
-            if unpacked_embedding_set["embeddings"] is not None
+            self._normalize_embeddings(unpacked_record_set["embeddings"])
+            if unpacked_record_set["embeddings"] is not None
             else None
         )
 
-        self._validate_embedding_set(
-            ids=unpacked_embedding_set["ids"],
-            embeddings=normalized_embeddings,
-            metadatas=unpacked_embedding_set["metadatas"],
-            documents=unpacked_embedding_set["documents"],
-            images=unpacked_embedding_set["images"],
-            uris=unpacked_embedding_set["uris"],
-            require_embeddings_or_data=require_embeddings_or_data,
+        unpacked_record_set["embeddings"] = normalized_embeddings
+
+        self._validate_record_set(
```

Comment:
This check can happen before we normalize or try to embed

Reply:
`normalization` is just unpacking from numpy or list - we should just move this logic to one place.